jcl/src/jdk.management/share/classes/com/ibm/lang/management/internal/ExtendedRuntimeMXBeanImpl.java

```diff
@@ -107,4 +108,19 @@ public boolean isVMIdle() {
 		}
 		return false;
 	}
+
+	@Override
+	public boolean isAttachApiInitialized() {
+		return AttachHandler.isAttachApiInitialized();
+	}
+
+	@Override
+	public boolean isAttachApiTerminated() {
+		return AttachHandler.isAttachApiTerminated();
+	}
+
+	@Override
+	public String getVmId() {
```

Comment:
This needs a security check.

Reply:
Clarification: `AccessController.doPrivileged()` or `SecurityManager.checkPermissions()`?

Reply:
SecurityManager.checkPermission(). When a security manager is enabled, untrusted code should not be able to call this API.